src/libraries/System.Text.Json/gen/JsonSourceGenerator.Roslyn4.0.cs

```diff
@@ -39,11 +36,11 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                     Parser.JsonSerializableAttributeFullName,
                     (node, _) => node is ClassDeclarationSyntax,
                     (context, _) => (ContextClass: (ClassDeclarationSyntax)context.TargetNode, context.SemanticModel))
-                .Combine(knownTypeSymbols)
                 .Select(static (tuple, cancellationToken) =>
                 {
-                    Parser parser = new(tuple.Right);
```

Comment:
note: based on what we saw with the options-generator, can you check this tuple *first* and not do *anything* in the case where the number of found nodes is 0?  that will avoid even doing things like loading symbols from teh compilation when it's entirely not necessary.  Tagging @tarekgh

Reply:
@CyrusNajmabadi just FYI, I have merged the suggested change in options source generator in main and opened PR https://github.com/dotnet/runtime/pull/93481 to backport it to 8.0 release.

Reply:
When it comes to symbol loading in the `KnownTypeSymbols` class it's all done lazily so an empty input set wouldn't force any.

Reply:
Come to think of it, given that this incremental value is subscribed to attribute annotations, I would expect that the number of nodes should always be greater than zero.